### PR TITLE
removed .data from test_autograd.py

### PR DESCRIPTION
**Pull request by anjali411**
issue: https://github.com/pytorch/pytorch/issues/33630

### CODE REVIEW

File: test/test_autograd.py [hunk 1]
```diff
@@ -295,28 +296,28 @@ def test_hessian_vector(self):
         z = x ** 2 + y * x + y ** 2
         z.backward(torch.ones(2, 2), create_graph=True)
 
-        x_grad = 2 * x.data + y.data
```
Comment (albanD) [inline test/test_autograd.py]: Potentially we want these to run in no_grad mode, but that doesn't really change anything give that they are just used in the assertEqual below...

File: test/test_autograd.py [hunk 2]
```diff
@@ -924,19 +925,20 @@ def test_indexing(self):
         y = Variable(x, requires_grad=True)
 
         def compare(x, y, idx, indexed_tensor, indexed_var):
-            indexed_var_t = indexed_var.data
+            indexed_var_t = indexed_var
```
Comment (albanD) [inline test/test_autograd.py]: The goal of this code is to ensure that indexing with a Tensor or a Variable gives the same result.
If we keep this test, then we should keep the `.data` here.
@gchanan should we keep this test though?

File: test/test_autograd.py [hunk 3]
```diff
@@ -1172,7 +1175,7 @@ def test_attribute_deletion(self):
         with self.assertRaises(RuntimeError):
             del x.data
         with self.assertRaises(TypeError):
-            x.data = None
+            x = None
```
Comment (albanD) [inline test/test_autograd.py]: That does not raise any error right? This should fail.
And you want to keep the original `.data` here as we explicitely want to check that its behavior did not change.

File: test/test_autograd.py [hunk 4]
```diff
@@ -3346,7 +3350,7 @@ class DeepReentrant(Function):
             @staticmethod
             def forward(ctx, x):
                 with torch.enable_grad():
-                    ctx.x = Variable(x.data, requires_grad=True)
+                    ctx.x = Variable(x, requires_grad=True)
```
Comment (albanD) [inline test/test_autograd.py]: You want a `.detach()` here I think. As you don't want `ctx.x` to be linked to `x` in a differentiable manner.

File: test/test_autograd.py [hunk 5]
```diff
@@ -3380,7 +3384,7 @@ class Reentrant(Function):
...
```
Comment (albanD) [inline test/test_autograd.py]: Same as above.

File: test/test_autograd.py [hunk 6]
```diff
@@ -3473,7 +3477,7 @@ def backward(ctx, input):
         torch.autograd.backward([t3])
 
         # Verify callback is called only once.
-        self.assertEquals(1, counter[0])
+        self.assertEqual(1, counter[0])
```
Comment (albanD) [inline test/test_autograd.py]: Why did that change?

**Review by albanD — CHANGES_REQUESTED** [review]
Looks mostly good, just small changes

File: test/test_autograd.py (hunk 6 above)
Comment (anjali411) [reply test/test_autograd.py]: assertEquals has been deprecated

**Review by albanD — APPROVED** [review]
Looks good !
